src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.Fixer.cs

```diff
@@ -21,9 +29,51 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
+            var diagnostic = context.Diagnostics[0];
+            var codeAction = CodeAction.Create("Use indexer", c => UseCollectionDirectly(context.Document, context.Span, c));
+            context.RegisterCodeFix(codeAction, diagnostic);
+
             // Fixer not yet implemented.
             return Task.CompletedTask;
 
         }
+
+        private async Task<Document> UseCollectionDirectly(Document document, TextSpan span, CancellationToken c)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(c).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(c).ConfigureAwait(false);
+
+            var invocationSyntax = GetSyntaxOfType<InvocationExpressionSyntax>(syntaxRoot.FindNode(span));
+            var invocationOp = (IInvocationOperation)semanticModel.GetOperation(invocationSyntax);
```

Comment:
This cast may not always succeed. Please use `as IInvocationOperation` instead.

Reply:
Done in the new commit.